packages/cli/src/runners/task-runner-process.ts

```diff
@@ -68,15 +68,14 @@ export class TaskRunnerProcess extends TypedEmitter<TaskRunnerProcessEventMap> {
 	) {
 		super();
 
-		a.ok(
-			this.runnerConfig.mode === 'internal_childprocess' ||
-				this.runnerConfig.mode === 'internal_launcher',
-		);
-
 		this.logger = logger.scoped('task-runner');
 	}
 
 	async start() {
+		a.ok(
+			this.runnerConfig.mode === 'internal_childprocess' ||
+				this.runnerConfig.mode === 'internal_launcher',
+		);
```

Comment:
To better understand - Why don't we want to fail immediately in the constructor if the mode is not as expected?

Reply:
This was because I noticed `N8N_RUNNERS_MODE=external` was broken. I could also extract this to a separate PR